uniffi_bindgen/src/lib.rs

```diff
@@ -115,6 +115,101 @@ use bindings::TargetLanguage;
 use interface::ComponentInterface;
 use scaffolding::RustScaffolding;
 
+pub trait BindingGeneratorConfig: for<'de> Deserialize<'de> + MergeWith + for<'a> From<&'a ComponentInterface> + Default + std::fmt::Debug {}
+
+/// A trait representing an external binding generator
+/// 
+/// # Example
+/// ```
+/// use uniffi_bindgen::{interface::ComponentInterface, MergeWith, BindingGeneratorConfig, BindingGenerator};
+/// use serde::Deserialize;
+/// use std::path::Path;
+/// 
+/// #[derive(Default, Debug, Deserialize)]
+/// pub struct KotlinConfig {}
+/// 
+/// impl MergeWith for KotlinConfig {
+///     fn merge_with(&self, _other: &KotlinConfig) -> Self {
+///         Self {} // This should implement the merging of two Configs, omitted for example
+///     }
+/// }
+/// 
+/// impl From<&ComponentInterface> for KotlinConfig {
+///     fn from(ci: &ComponentInterface) -> Self {
+///         Self {} // This should implement generating a config from a ComponentInterface,
+///                 // omitted for example
+///     }
+/// }
```

Comment:
I don't think this is new, but this From implementation feels confusing to me.  What if this was named `BindingGeneratorConfig::new(ci: &ComponentInterface)` instead?